src/farcasterd/trade_state_machine.rs

```diff
@@ -768,7 +947,11 @@ fn attempt_transition_from_swapd_launched_to_swapd_running(
         }
         (BusMsg::Ctl(CtlMsg::Hello), ServiceId::Peer(..)) => {}
         _ => {
-            trace!("BusMsg {} invalid for state swapd launched", event.request);
+            trace!(
+                "{} | BusMsg {} invalid for state swapd launched",
+                swap_id,
+                event.request
+            );
         }
     }
     let peerd_up = runtime.registered_services.contains(&peerd);
```

Comment:
with a unique `ServiceId` for every peerd instantiation, if we create a new peerd instance for a remote and kill the old one, this will return `false` since the old peerd's `ServiceId` is cached in the tsm but won't be in the `runtime.registered_services` anymore, so swap will never launch. So we can instead here check only against the node address of the peerd key in `runtime.registered_services`, but if there are multiple concurrent peerds living for said remote, then still should ensure that we launch with the latest one.

Reply:
I've pushed a commit that does an ugly impl of this. TODO if we go with this change:
1. it currently just uses the first peerd it finds in registered services - imo we'd need to encode creation time in the peerd `ServiceId` to do that correctly
2. the `SwapdLaunched` struct should not keep the full peerd `ServiceId` since only the `NodeAddr` is read with this change